src/main/java/net/neoforged/neoforge/common/extensions/IItemStackExtension.java

```diff
@@ -83,11 +82,9 @@ default int getBurnTime(@Nullable RecipeType<?> recipeType) {
         if (self().isEmpty()) {
             return 0;
         }
-        // TODO 1.20.5: Throw if the returned value from the item is negative, and move datamap logic to the item method.
         int burnTime = self().getItem().getBurnTime(self(), recipeType);
         if (burnTime < 0) {
-            FurnaceFuel fuel = self().getItem().builtInRegistryHolder().getData(NeoForgeDataMaps.FURNACE_FUELS);
-            burnTime = fuel != null ? fuel.burnTime() : 0;
+            throw new IllegalStateException("Stack of item " + BuiltInRegistries.ITEM.getKey(self().getItem()) + "has a negative burn time");
```

Comment:
Missing space between item name and appended string.
```suggestion
            throw new IllegalStateException("Stack of item " + BuiltInRegistries.ITEM.getKey(self().getItem()) + " has a negative burn time");
```